### PR TITLE
Monero wallet network, Testnet

### PR DESCRIPTION
**Pull request by da-kami**
Add network for monero and switch application to run on Bitcoin `testnet3` and Monero `stagenet`.

### CODE REVIEW

File: xmr-btc/src/bob.rs [hunk 1]
```diff
@@ -901,6 +901,8 @@ impl State5 {
 
         // NOTE: This actually generates and opens a new wallet, closing the currently
         // open one.
+        // TODO: This means that the wallet-rpc HAS to be started with the --wallet-dir
```
Comment (D4nte) [inline xmr-btc/src/bob.rs:904]: I don't think this TODO is helpful.

File: xmr-btc/src/config.rs [hunk 2]
```diff
@@ -26,6 +27,23 @@ impl Config {
...
             bitcoin_network: ::bitcoin::Network::Bitcoin,
+            monero_network: ::monero::Network::Mainnet,
+        }
+    }
+
+    pub fn testnet() -> Self {
+        Self {
+            bob_time_to_act: *mainnet::BOB_TIME_TO_ACT,
+            bitcoin_finality_confirmations: mainnet::BITCOIN_FINALITY_CONFIRMATIONS,
+            bitcoin_avg_block_time: *mainnet::BITCOIN_AVG_BLOCK_TIME,
```
Comment (D4nte) [inline xmr-btc/src/config.rs:38]: This is not correct, can you please specify the actual values?

**Review by D4nte — APPROVED** [review]

**Comment by da-kami** [comment]
Note: I would keep the logging of `monero_harness` and `bitcoin_harness` for now, but would change the log-level in `main` to `info` (currently set to `trace`...). I think we should see the harness logs, but the log level on master should be corrected. See my last commit for that :)

**Review by D4nte — APPROVED** [review]

**Comment by da-kami** [comment]
With the last commit the problem of Alice not waiting for Bitcoin tx finality before locking Monero is fixed. Tested on test/stagenet.

**Comment by da-kami** [comment]
bors cancel

**Pull request merged**